v3/src/components/graph/components/scatter-plot-utils.ts

```diff
@@ -44,6 +51,37 @@ export function scatterPlotFuncs(layout: GraphLayout, dataConfiguration?: IGraph
     return { xValue, yValue, xCoord, yCoord, rightCoord, color }
   }
 
+  function connectingLine(caseID: string, plotNum: number) {
+    const dataset = dataConfiguration?.dataset
+    const xValue = getXCoord(caseID)
+    const yValue = getYCoord(caseID, plotNum)
+    if (isFinite(xValue) && isFinite(yValue)) {
+      const caseData = dataset?.getCase(caseID)
+      if (caseData) {
+        const lineCoords: [number, number] = [xValue, yValue]
+        return {
+          caseData,
+          lineCoords,
+          plotNum
+        }
+      }
+    }
+  }
+
+  function connectingLinesForCases() {
+    const lineDescriptions: IConnectingLineDescription[] = []
+    const dataset = dataConfiguration?.dataset
+    dataset?.cases.forEach(c => {
+      const line = connectingLine(c.__id__, 0)
+      line && lineDescriptions.push(line)
+      const y2line = yAttrIDs[1] && connectingLine(c.__id__, 1)
+      y2line && lineDescriptions.push(y2line)
+      const yRightLine = yAttrIDs[2] && connectingLine(c.__id__, 2)
```

Comment:
I'm confused by these hard-coded indices given that there can be an arbitrary number of y attributes.

Reply:
Thanks for pointing that out. I think I've improved/fixed this with the latest commit.